conans/model/conan_file.py

```diff
@@ -111,10 +111,12 @@ def __init__(self, display_name=""):
         self.env_info = MockInfoProperty("env_info")
         self._conan_dependencies = None
 
-        if not hasattr(self, "virtualbuildenv"):  # Allow the user to override it with True or False
-            self.virtualbuildenv = True
-        if not hasattr(self, "virtualrunenv"):  # Allow the user to override it with True or False
-            self.virtualrunenv = True
+        # Allow the user to override them with True or False
+        # If not, value will be given in profile_node_definer based on confs
+        if not hasattr(self, "virtualbuildenv"):
```

Comment:
This might be breaking behaviour if we consider that anyone could have been doing `virtualbuildenv = None` in their conanfile to inhibit it, instead of setting it to `False` or something, but I think it's acceptable?

Reply:
I think these are not documented yet? If not documented it might be more than acceptable the risk.